x11/turbovnc-viewer/Portfile

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           cmake 1.1
+PortGroup           github 1.0
+PortGroup           java 1.0
+
+github.setup        TurboVNC turbovnc 2.2.6
+
+name                turbovnc-viewer
+platforms           darwin
+license             GPL-2
+maintainers         {gmail.com:audvare @Tatsh}
+categories          x11 net
+description         TurboVNC VNC viewer.
+long_description    ${description}
+
+checksums           rmd160  de684a6e9c2f2e1ef1f05b98229fd40269d17a7c \
+                    sha256  71a89600d1e4ffe67aa65e5ddc6b21b1659b9105892c07074e99181218dc33e8 \
+                    size    9023665
+
+depends_lib         port:libjpeg-turbo
+
+java.version        1.8+
+
+# This must be called to set ${java.home}
+java_set_env
+configure.args \
+    -DTVNC_BUILDSERVER=no \
+    -DCMAKE_C_FLAGS=-I${prefix}/include \
+    -DTJPEG_JAR=${prefix}/share/java/turbojpeg.jar \
+    -DJAVA_INCLUDE_PATH=${java.home}/include \
+    -DJAVA_INCLUDE_PATH2=${java.home}/include/darwin \
+    -DJAVA_AWT_INCLUDE_PATH=${java.home}/include \
+    -DTVNC_BUILDNATIVE=ON
+
+patch {
+    reinplace "s|@VERSION@|${version}|g" ${worksrcpath}/release/Info.plist.in
+    reinplace "s|@PKGVENDOR@|MacPorts|g" ${worksrcpath}/release/Info.plist.in
+    reinplace "s|@BUILD@|${version}|g" ${worksrcpath}/release/Info.plist.in
+    reinplace "s|@CMAKE_PROJECT_NAME@|TurboVNC|g" ${worksrcpath}/release/Info.plist.in
+    reinplace "s|^<dict>|<dict><key>NSHighResolutionCapable</key><true/>|" ${worksrcpath}/release/Info.plist.in
+}
+
+post-destroot {
+    set appdir "${destroot}${applications_dir}/TurboVNC Viewer.app"
+    xinstall -m 755 -d ${appdir}/Contents/MacOS
+    xinstall -m 755 -d ${appdir}/Contents/Resources/Java
+    xinstall -m 755 -d ${appdir}/Contents/Resources/Native
+    xinstall -m 644 ${worksrcpath}/release/Info.plist.in \
+        ${appdir}/Contents/Info.plist
+    xinstall -m 755 ${worksrcpath}/unix/vncviewer/JavaAppLauncher \
+        "${appdir}/Contents/MacOS/TurboVNC Viewer"
+    set chan [open ${appdir}/Contents/PkgInfo w+]
+    puts -nonewline ${chan} "APPL????"
+    close ${chan}
+    xinstall -m 644 ${worksrcpath}/release/turbovnc.icns \
+        ${worksrcpath}/release/vncviewer.icns ${appdir}/Contents/Resources/
+    xinstall -m 755 ${prefix}/lib/libturbojpeg.dylib \
+        ${cmake.build_dir}/java/libturbovnchelper.dylib \
+        ${appdir}/Contents/Resources/Native/
+    xinstall -m 644 ${cmake.build_dir}/java/VncViewer.jar \
+        ${appdir}/Contents/Resources/Java/
+}
+
+if {![file exists ${java.home}/include/jni.h]} {
+    depends_lib-delete openjdk11
+    pre-configure {
+        known_fail yes
+        ui_error "
+****
+**** ${name} requires port openjdk11 with variant +JNI.
```

Comment:
if you require a port to be installed with a certain variant you should use the `PortGroup active_variants 1.1`; you ca look for examples in other Portfiles

Reply:
Thanks. Fixed with latest push.